engine/src/parameters.rs

```diff
@@ -130,13 +130,48 @@ impl SubmitResult {
     }
 }
 
-/// Borsh-encoded parameters for the `call` function.
-#[derive(BorshSerialize, BorshDeserialize)]
+/// Borsh-encoded parameters for the engine `call` function.
+#[derive(BorshSerialize, BorshDeserialize, Debug, PartialEq, Eq, Clone)]
 pub struct FunctionCallArgs {
+    pub contract: RawAddress,
+    /// Wei compatible Borsh-encoded value field to attach an ETH balance to the transaction
+    pub value: WeiU256,
+    pub input: Vec<u8>,
+}
+
+/// Legacy Borsh-encoded parameters for the engine `call` function, to provide backward type compatibility
+#[derive(BorshSerialize, BorshDeserialize, Debug, PartialEq, Eq, Clone)]
+pub struct FunctionCallArgsLegacy {
     pub contract: RawAddress,
     pub input: Vec<u8>,
 }
 
+/// Deserialized values from bytes to current or legacy Borsh-encoded parameters
+/// for passing to the engine `call` function, and to provide backward type compatibility
+#[derive(BorshSerialize, BorshDeserialize, Debug, PartialEq, Eq, Clone)]
+pub enum CallArgs {
+    New(FunctionCallArgs),
```

Comment:
`New` -> `V2`
`Legacy` -> `V1`

Reply:
OK, fixed here https://github.com/aurora-is-near/aurora-engine/pull/351/commits/8e1306750eed1424c19f05d5ff40bfa61a6d0380